app/collections/settings.jsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import alert from '../lib/alert';
+import DisplayNameSlugEditor from '../partials/display-name-slug-editor';
+
+export default class CollectionSettings extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      error: null,
+      isDeleting: false,
+      setting: {
+        private: false
+      }
+    };
+
+    this.confirmDelete = this.confirmDelete.bind(this);
+    this.deleteCollection = this.deleteCollection.bind(this);
+    this.handleToggle = this.handleToggle.bind(this);
+    this.redirect = this.redirect.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.collection.listen('delete', this.redirect);
+  }
+
+  componentDidUnMount() {
+    this.props.collection.stopListening('delete');
+  }
+
+  redirect() {
+    this.context.router.push('/collections');
+  }
+
+  confirmDelete() {
+    alert((resolve) => {
+      const handleDelete = () => { this.deleteCollection().then(resolve); };
+
+      return (
+        <div className="confirm-delete-dialog content-container">
+          <p>Are you sure you want to delete this collection? This action is irreversible!</p>
+
+          {this.state.isDeleting &&
+            <div>
+              <button className="major-button" disabled={true}><i className="fa fa-spinner" /></button>
+              {' '}
+            </div>}
+
+          {!this.state.isDeleting &&
+            <div>
+              <button className="major-button" onClick={handleDelete}>Yes, delete it!</button>
+              {' '}
+              <button className="minor-button" onClick={resolve}>No, don't delete it.</button>
+            </div>}
```

Comment:
Added it to the cancel since that made sense to me. But could add it to the confirm button instead.